realm/src/main/java/io/realm/AsyncRealmQueryResult.java

```diff
@@ -0,0 +1,25 @@
+package io.realm;
+
+import java.util.concurrent.Future;
+
+/**
+ * Represents the result of an asynchronous Realm query.
+ *
+ * Users are responsible of maintaining a reference to {@code AsyncRealmQueryResult} in order
+ * to call #cancel in case of a configuration change for example (to avoid memory leak, as the
+ * query will post the result to the caller's thread callback)
+ */
+public class AsyncRealmQueryResult {
```

Comment:
Shouldn't this be called ```AsyncRealmQueryRequest``` instead as you don't actually have any result yet?
Also, I am not sure I am a big fan of having this as a top level class as having Realm as part of the name seems sorta long to me.

Maybe as a Inner class of RealmQuery, like ```RealmQuery.Request``` or ```RealmQuery.AsyncRequest``` ?